[[ Merge 6.7.6 rc 1]]

livecodesebastien opened the pull request:
The only non-reviewed are the changes to the player, with had to be refactored.

livecodepanos committed:
[[ Bug 14435 ]] Allow setting a filename path that is relative to the defaultFolder

livecodepanos committed:
Check if player filename is successfully resolved

to prevent a crash in case the filename does not exist

livecodepanos committed:
Merge remote-tracking branch 'upstream/develop-6.7' into bugfix-14435

livecodepanos committed:
[Bug 14435] Check if resolveplayerfilename() is successful

livecodepanos committed:
Merge remote-tracking branch 'upstream/develop-6.7' into bugfix-14435

livecodepanos committed:
[Bug 14435] Store the player's resolved filename and use it in prepare()

livecodepanos committed:
Merge remote-tracking branch 'upstream/develop-6.7' into bugfix-14435

livecodepanos committed:
Updates after feedback

livecodepanos committed:
[Bug 14435] Resolve player filename in prepare()

to match the pre-LC6.7 behaviour

livecodepanos committed:
Merge branch 'develop-6.7' into bugfix-14435

Commit:
[[ TemplateAppMetadata ]] Regression fix on custom android template manifest and implementation of custom template plist for iOS

Conflicts:
	ide-support/revsaveasiosstandalone.livecodescript
	prebuilt

livecodeian committed:
[[ Bug 15324 ]] Convert image pixel format before sending to custom printer

runrevmark committed:
[[ Bug ]] Make sure MCUIDC initializes all fields to defaults (as it used as is in server mode).

livecodestephen committed:
[[ Bug ]] Delete the m_fonts list in MCNewFontlist::~MCNewFontlist()

livecodepanos committed:
[Bug 14349] Subclass QTMovieView and override hitTest method

to allow mouse events pass to the superview

livecodesebastien committed:
[[ Bug 14466 ]] Force a call to readsome when reading from sockets

livecodesebastien committed:
[[ AndroidVersions ]] Make the update of the Android minimum version possible from script-only stacks

livecodepanos committed:
[Bug 14349] Send mouseDown msg after mdown is passed to the controller

Also make sure you include the QTKit framework in settings

livecodepanos committed:
Added bugfix note

livecodepanos committed:
[Bug 15372] wait with messages cause crash on Mac OS

livecodepanos committed:
Updates after feedback

livecodesebastien committed:
[[ DeployAllPlatforms ]] Enable standalone deployment from the repo for non-native platforms, if the binaries have been copied

livecodesebastien committed:
[[ Bug 15385 ]] Enqueue the same message for keyDown/keyUp when dead chars fail to combine

livecodesebastien committed:
Merge pull request #2290 from livecodesebastien/bugfix-15385

[[ Bug 15385 ]] Enqueue the same message for keyDown/keyUp when dead …

livecodesebastien committed:
Merge pull request #2280 from runrevpanos/bugfix-15372

[Bug 15372] wait with messages cause crash on Mac OS

livecodesebastien committed:
Merge pull request #2274 from runrevpanos/bugfix-14349

[Bug 14349] Can't use mouse events on QTKit player

livecodesebastien committed:
Merge pull request #2269 from livecodesebastien/bugfix-14466_newfix

[[ Bug 14466 ]] Force a call to readsome when reading from sockets

livecodesebastien committed:
Merge pull request #2257 from runrev/bugfix-template_app_metadata

[[ TemplateAppMetadata ]] Regression fix on custom android template m…

livecodesebastien committed:
Merge pull request #1716 from runrevpanos/bugfix-14435

[[ Bug 14435 ]] Allow setting a filename path that is relative to the de...

runrevmark committed:
[[ Bug 15392 ]] Make sure we ignore any trailing returns when setting the points of a graphic.

livecodesebastien committed:
[[ Bug 15401 ]] Restore the defaultFolder to its value before returning from deployIsValidSDK

runrevmark committed:
[[ RemoveThreadedRendering ]] Remove all code related to threaded rendering.

runrevmark committed:
[[ Bug ]] Incorrect field used in GCacheTable causing leak of key.

runrevmark committed:
[[ Bug 15408 ]] Deleted controls can still be referenced by id.

When a control was deleted by script, only the id cache for that object was being flushed leaving any child objects in the id cache.

This has been fixed by making 'scheduledelete' virtual with a parameter to determine whether it is for a child, or direct. This then recurses to scheduledelete each child.

livecodesebastien committed:
Merge pull request #2304 from livecodesebastien/bugfix-15401

[[ Bug 15401 ]] Restore the defaultFolder to its value before returni…

livecodesebastien committed:
Merge pull request #2293 from runrevmark/bugfix-15392

[[ Bug 15392 ]] Make sure we ignore any trailing returns when setting…

livecodesebastien committed:
Merge pull request #2282 from livecodesebastien/feature-deploy_all_platforms

[[ DeployAllPlatforms ]] Enable standalone deployment from the repo f…

livecodesebastien committed:
Merge pull request #2262 from runrevian/bugfix-15324

[[ Bug 15324 ]] Convert image pixel format before sending to custom p…

livecodesebastien committed:
Merge pull request #2273 from livecodesebastien/android_versions_reworked

[[ AndroidVersions ]] Make the update of the Android minimum version …

livecodesebastien committed:
Merge pull request #2307 from runrevmark/bugfix-valgrind-6_7

[[ Bug ]] Fixes for issues found via valgrind (6.7)

livecodesebastien committed:
Merge pull request #2306 from runrevmark/remove-threaded_rendering

[[ RemoveThreadedRendering ]] Remove all code related to threaded ren…

livecodesebastien committed:
Merge pull request #2311 from runrevmark/bugfix-15408

[[ Bug 15408 ]] Deleted controls can still be referenced by id.

livecodesebastien committed:
Merge remote-tracking branch 'origin/develop-6.7' into develop-7.0

Conflicts:
	engine/Android.mk
	engine/Makefile.kernel
	engine/Makefile.kernel-server
	engine/engine-mobile.xcodeproj/project.pbxproj
	engine/engine.xcodeproj/project.pbxproj
	engine/kernel.vcproj
	engine/src/desktop.cpp
	engine/src/field.cpp
	engine/src/globals.cpp
	engine/src/globals.h
	engine/src/graphic.cpp
	engine/src/lnxflst.cpp
	engine/src/lnxgtktheme.cpp
	engine/src/object.h
	engine/src/player-platform.cpp
	engine/src/player-platform.h
	engine/src/stack.h
	engine/src/sysdefs.h
	engine/src/util.cpp
	ide
	ide-support/revsaveasiosstandalone.livecodescript
	libgraphics/src/lnxtext.cpp

livecodesebastien committed:
[[ LC 7.0.6 RC 1 ]] Update ide submodule update

livecodesebastien committed:
[[ iOS Deployment Broken ]] TextEncode does not exist in LC 6.7

livecodesebastien committed:
[[ Merge 6.7.6 RC 1 ]] Fix compile issues

livecodesebastien committed:
[[ Merge 6.7.6 RC 1 ]] Remove stacktile.h, stacktile.cpp, systhread.h and sysw32thread.cpp from kernel.vcproj

livecodesebastien committed:
[[ Merge 6.7.6 RC 1 ]] Fix Mac compiltation issues

runrevmark committed:
[[ Bug 15113 ]] Make sure default buttons never lose the glow whilst the app is active on Yosemite.

runrevmark commented:
There's a problem here due to subclassing a QTKit class - we can't do that as we mustn't link to QTKit directly.

livecodesebastien committed:
Revert "Merge pull request #2274 from runrevpanos/bugfix-14349"

This reverts commit 35233e74f9ae5f2d662d17029a8532acba2aec22, reversing
changes made to 6ecb9b793fe05d5838b660ef74de5b7c72d63feb.

Conflicts:
	engine/engine.xcodeproj/project.pbxproj

livecodesebastien committed:
Merge pull request #2314 from livecodesebastien/regression-ios_deployment_broken

[[ iOS Deployment Broken ]] TextEncode does not exist in LC 6.7

livecodesebastien committed:
Merge pull request #2318 from livecodesebastien/revert-bugfix-14349

[[ Revert bugfix 14349 ]]

livecodepanos committed:
[Bug 14349] Dynamically subclass QTMovieView

to avoid adding QTKit to the required frameworks

livecodepanos committed:
[Bug 14349] dynamicallySubclassQTMovieView is now a class method

thus no need to create an instance of MCQTKitHelper

livecodesebastien committed:
Merge pull request #2320 from runrevpanos/bugfix-14349_new

[Bug 14349] Dynamically subclass QTMovieView

peter-b committed:
tools: Add a helper script for setting up Xcode for LiveCode builds

Add a script that installs all of the SDK symlinks needed for
building LiveCode into a Xcode app bundle (and checks whether
they're all available, of course!)

livecodesebastien committed:
Merge pull request #2324 from peter-b/misc-xcodesetup

tools: Add a helper script for setting up Xcode for LiveCode builds

livecodesebastien committed:
Merge pull request #2316 from runrevmark/bugfix-15113

[[ Bug 15113 ]] Make sure default buttons never lose the glow whilst …

livecodesebastien committed:
Merge remote-tracking branch 'origin/develop-7.0' into merge-6.7.6-rc-1

livecodesebastien committed:
Merge remote-tracking branch 'origin/develop-6.7' into merge-6.7.6-rc-1

Conflicts:
	engine/engine.xcodeproj/project.pbxproj

livecodesebastien committed:
[[ LC 6.7.6 RC 1 ]] Update ide submodule

livecodesebastien committed:
[[ LC 6.7.6 RC 1 ]] Bump version number

livecodepanos committed:
[LC 6.7.6-rc-1] Corrected typo in version number

livecodesebastien committed:
Merge remote-tracking branch 'origin/develop-6.7' into merge-6.7.6-rc-1

Conflicts:
	ide
	version

livecodeali commented on engine/src/stack.cpp:
```diff
@@ -3133,6 +3124,28 @@ bool MCStack::resolve_relative_path(MCStringRef p_path, MCStringRef& r_resolved)
+	
+    MCS_getcurdir(&t_cur_dir);
+
+    return MCStringFormat(r_resolved, "%@/%*@", *t_cur_dir, MCRangeMake(t_start_index, MCStringGetLength(p_path) - t_start_index), p_path);
```
The range argument here needs to be a pointer to an MCRange rather than an MCRange.

livecodesebastien committed:
[[ Merge 6.7.6 RC 1 ]] Give a MCRange pointer to MCStringFormat

livecodeali commented:
@livecode-vulcan review ok f11fec6

livecodeali commented:
@livecode-vulcan review ok fa20608

livecode-vulcan commented:
💙 review by @livecodeali ok fa20608a88ede892658a716cdbfefdcae22feb1b
